### PR TITLE
[#12] [#15] [Integrate] Navigate to other screens from the Login screen

### LINKED ISSUE #12: [UI] As a user, I can see the home screen after login success
## Why

After the user login success, navigate to the home screen as the landing page.

## Acceptance criteria

Show the home screen with:

- Current date information on the top left corner.
- `Today` title under the current date information.
- User profile image on the top right corner.

## Design

- [Design for the home screen](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=44%3A148&scaling=scale-down)

### PR DESCRIPTION
**Pull request by nmint8m**
- Close #12 and #15

## What happened 👀

This PR is to navigate from the Login screen to the Forget Password screen and the Home screen.

## Insight

In this PR, I created the singleton of `NavigationController`, which manages the navigation of the whole app. I also applied navigating from the Login Screen to the Forget Password screen and to the Home screen.

To prevent the user from navigating back to the Login screen by tapping on the physical back button, I used the `WillPopScope` with `onWillPop: () async => false`.

### CODE REVIEW

File: lib/utils/navigation_controller.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:injectable/injectable.dart';
+import 'package:kayla_flutter_ic/di/di.dart';
+
+final navigationController = getIt.get<NavigationController>();
+
+@singleton
+class NavigationController with ChangeNotifier {
```
Comment (manh-t) [inline lib/utils/navigation_controller.dart]: @nmint8m Can I know why we create a `ChangeNotifier` class for this controller? I think `Navigator` would work fine without `ChangeNotifier`, or you may want to store the route locally for a purpose 🤔

Comment (nmint8m) [reply lib/utils/navigation_controller.dart]: Oh sorry, I will remove this `ChangeNotifier`. Firstly I wanted to adopt [this implementation from Medium](https://medium.com/flutter/learning-flutters-new-navigation-and-routing-system-7c9068155ade) by using the **Navigator 2.0 API**. But it consumes the time that I mentioned [here](https://nimble-co.slack.com/archives/C04HDQ53E74/p1676265517230049). I will try that approach later.

**Comment by doannimble** [comment]
@nmint8m We don't have `Inisght` for this PR? And the `What happened` is vague. Could you please update?

**Comment by nmint8m** [comment]
@doannimble I updated the Insight 👍  Please check.

File: lib/utils/navigation_controller.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import 'package:injectable/injectable.dart';
+import 'package:kayla_flutter_ic/di/di.dart';
+
+final navigationController = getIt.get<NavigationController>();
+
+@singleton
+class NavigationController {
+  void push(BuildContext context, Widget widget) {
+    Route route =
+        MaterialPageRoute(builder: (context) => ProviderScope(child: widget));
```
Comment (manh-t) [inline lib/utils/navigation_controller.dart]: we just need to pass `ProviderScope` once in `main.dart` : `runApp(ProviderScope(child: MyApp()));`. Then we can get rid of `ProviderScope` here

**Review by manh-t — COMMENTED** [review]
rest LGTM 👏

File: lib/utils/navigation_controller.dart (hunk 2 above)
Comment (nmint8m) [reply lib/utils/navigation_controller.dart]: Error when removing the ProviderScope.

**Review by manh-t — APPROVED** [review]

**Pull request merged**